internal/provider/networks_wireless_ssids_resource.go

```diff
@@ -0,0 +1,1727 @@
+									Optional:            true,
+									CustomType:          jsontypes.StringType,
+								},
+								"secret": schema.StringAttribute{
+									MarkdownDescription: "Enable or disable the interface (only for MX devices). Valid values are 'enabled', 'disabled', and 'not configured'.",
+									Optional:            true,
+									CustomType:          jsontypes.StringType,
+								},
+								"ca_certificate": schema.StringAttribute{
+									MarkdownDescription: "Enable or disable the interface (only for MX devices). Valid values are 'enabled', 'disabled', and 'not configured'.",
+									Optional:            true,
+									CustomType:          jsontypes.StringType,
+								},
+								"port": schema.Int64Attribute{
+									MarkdownDescription: "The VLAN that management traffic should be tagged with. Applies whether usingStaticIp is true or false.",
+									Optional:            true,
+									CustomType:          jsontypes.Int64Type,
+								},
+								"radsec_enabled": schema.BoolAttribute{
+									MarkdownDescription: "Configure the interface to have static IP settings or use DHCP.",
+									Optional:            true,
+									CustomType:          jsontypes.BoolType,
+								},
+							},
+						},
+					},
+					"ap_tags_and_vlan_ids": schema.SetNestedAttribute{
+						Optional: true,
+						Computed: true,
+						NestedObject: schema.NestedAttributeObject{
+							Attributes: map[string]schema.Attribute{
+								"vlan_id": schema.Int64Attribute{
+									MarkdownDescription: "The VLAN that management traffic should be tagged with. Applies whether usingStaticIp is true or false.",
+									Optional:            true,
+									CustomType:          jsontypes.Int64Type,
+								},
+								"tags": schema.ListAttribute{
+									MarkdownDescription: "Up to two DNS IPs.",
+									Optional:            true,
+									ElementType:         jsontypes.StringType,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (r *NetworksWirelessSsidsResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*openApiClient.APIClient)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected *http.Client, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.client = client
+}
+
+func (r *NetworksWirelessSsidsResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var data *NetworksWirelessSsidsResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	request := openApiClient.NewUpdateNetworkWirelessSsidRequest()
```

Comment:
So it looks like the Meraki API only has a GET and a POST for this resource. That means we are leveraging the same API call for the Create, Update and Delete funcs. 

In cases like this we need to reduce repeated code as much as possible so rather than defining the same payload over in each one we can create a function that assembles the payload for us:

```
func NetworksWirelessSsidsPayload(ctx context.Context, data *NetworksWirelessSsidsResourceModel) (openApiClient.UpdateNetworkWirelessSsidRequest, diag.Diagnostics) {
	payload := *openApiClient.NewUpdateNetworkWirelessSsidRequest()

	return payload, nil
}
```

This can be done with the response as well if needed:

```
func NetworksWirelessSsidsResponse(ctx context.Context, response *openApiClient.GetNetworkWirelessSsids200ResponseInner, data *NetworksWirelessSsidsResourceModel) (*NetworksWirelessSsidsResourceModel, diag.Diagnostics) {

	return data, nil
}
```